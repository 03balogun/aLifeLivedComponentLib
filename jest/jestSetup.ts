/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
