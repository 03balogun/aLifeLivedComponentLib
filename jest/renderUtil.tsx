import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
