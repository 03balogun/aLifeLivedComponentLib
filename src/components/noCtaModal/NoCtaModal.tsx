import React from 'react';

// Components
import { Modal } from '../../themes/modal/Modal';
import { ModalContent } from '../../themes/modal/ModalContent';
import { ModalHeader } from '../../themes/modal/ModalHeader';

type ProfileModalProps = {
  onClose: () => void;
  isVisible: boolean;
  header: string;
  description: string;
};

export const NoCtaModal: React.FC<ProfileModalProps> = ({
  onClose,
  isVisible,
  header,
  description,
}) => {
  return (
    <Modal onClose={() => onClose()} isVisible={isVisible}>
      <ModalHeader onHeaderPress={() => onClose()} />
      <ModalContent header={header} description={description} />
    </Modal>
  );
};
