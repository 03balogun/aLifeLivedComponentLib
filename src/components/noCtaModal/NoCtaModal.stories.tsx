import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { NoCtaModal } from './NoCtaModal';

storiesOf('Modal', module)
  .addDecorator(Story => Story())
  .add('profile', () => (
    <NoCtaModal
      isVisible={true}
      onClose={action('On close action fired')}
      header="Welcome to the profile page! 🙋‍♀️🙋"
      description="You can see all the stories you have created here. As well as the stories you have liked or bookmarked!"
    />
  ))
  .add('search', () => (
    <NoCtaModal
      isVisible={true}
      onClose={action('On close action fired')}
      header="Welcome to the search page! 🔎"
      description="Here you can search for stories by title or tags"
    />
  ))
  .add('recording-success', () => (
    <NoCtaModal
      isVisible={true}
      onClose={action('On close action fired')}
      header="Success! 👏"
      description="You've completed your first recording! Now to keep going closing this modal and pressing the 'Proceed to next question' button"
    />
  ));
