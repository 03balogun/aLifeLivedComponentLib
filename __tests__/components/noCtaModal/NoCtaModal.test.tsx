import React from 'react';
import { render, fireEvent } from './../../../jest/renderUtil';

// Component
import { NoCtaModal } from '../../../src/components/noCtaModal/NoCtaModal';

const mockOnClose = jest.fn();

const setup = (isVisible = true) => {
  const utils = render(
    <NoCtaModal
      isVisible={isVisible}
      onClose={mockOnClose}
      header="Welcome to the no cta modal!"
      description="This is a modal that has no cta buttons and is just text!"
    />
  );

  return {
    ...utils,
  };
};

describe('component >> noCtaModal', () => {
  describe('Modal is visible', () => {
    test('Should render no cta modal correctly', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should render correct text', () => {
      const { getByText } = setup();
      expect(getByText('Welcome to the no cta modal!')).not.toBeNull();
      expect(
        getByText('This is a modal that has no cta buttons and is just text!')
      ).not.toBeNull();
    });

    test('Should fire modal close', () => {
      const { getByTestId } = setup();
      fireEvent.press(getByTestId('modal-close-icon'));
      expect(mockOnClose).toBeCalledTimes(1);
    });
  });

  describe('Modal is NOT visible', () => {
    test('not render modal', () => {
      const { toJSON } = setup(false);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
