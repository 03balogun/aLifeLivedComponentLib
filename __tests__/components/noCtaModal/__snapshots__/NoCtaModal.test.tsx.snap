// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component >> noCtaModal Modal is NOT visible not render modal 1`] = `null`;

exports[`component >> noCtaModal Modal is visible Should render no cta modal correctly 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 100,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#252424",
          "borderRadius": 8,
          "display": "flex",
          "margin": 8,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "flex-start",
            "alignItems": "flex-start",
            "display": "flex",
            "margin": 8,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f4f5f5",
                "display": "flex",
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 24,
              },
            ]
          }
        >
          X
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "display": "flex",
            "margin": 16,
            "paddingBottom": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#f4f5f5",
              "display": "flex",
              "fontSize": 24,
              "fontWeight": "bold",
              "lineHeight": 30,
              "textAlign": "center",
            },
          ]
        }
      >
        Welcome to the no cta modal!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#f4f5f5",
              "display": "flex",
              "fontSize": 16,
              "lineHeight": 20,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        This is a modal that has no cta buttons and is just text!
      </Text>
    </View>
  </View>
</View>
`;
